src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -764,6 +790,15 @@ def update_ssh_access(self, agentpool: AgentPool) -> AgentPool:
             agentpool.security_profile.ssh_access = ssh_access
         return agentpool
 
+    def set_up_pod_ip_allocation_mode(self, agentpool: AgentPool) -> AgentPool:
```

Comment:
I suppose you don't need this in AKSPreviewAgentPool**Update**Decorator, from help msg, the option is only supported in `aks create` and `aks nodepool add`, but  AKSPreviewAgentPool**Update**Decorator works for `aks update` and `aks nodepool update`.

Reply:
Removed the definition method as well from the AgentPoolUpdateDecorator.